driver/Daruma.py

```diff
@@ -148,26 +148,33 @@ def add_missing_provider(self, missing_provider):
         logger.debug("adding a missing provider")
         if not self.file_manager.add_missing_provider(missing_provider):
             return False
-        # this was a correct missing provider; we can add it to the bootstrap_manager
-        self.bootstrap_manager.providers = self.file_manager.providers
+        # this was a correct missing provider; we can use it
+        self._set_all_internal_providers(self.file_manager.providers)
 
         return True
 
+    def _set_all_internal_providers(self, providers):
+        """
+        Sets the internal provider objects for all submanagers
+        """
+        self.file_manager.providers = providers
+        self.bootstrap_manager.providers = providers
+        self.resilience_manager.providers = providers
+
     def _reset(self):
         """
         Reprovision the system
         To be called after a core parameter change (change in provider or threshold)
         Raises FatalOperationFailure if there was an unrecoverable write error
         """
         logger.debug("_resetting")
-        self._load_manifest()
         try:
             self.file_manager.reset()
             self.update_master_key()
             self.resilience_manager.log_success()
         except exceptions.OperationFailure as e:
-            # don't try to repair - if we are here, all files got successfully refreshed
-            self.resilience_manager.diagnose(e.failures)
+            # this should never occur
```

Comment:
modified file_manager.reset so that it never throws opfails. the issue is that if we are reprovisioning away from some set of providers, as long as we can get files off the old set, we don't care about punishing or repairing them. and update_master_key never threw opfails anyway, so we should never get an opfail here